Tribler/community/tunnel/hidden_community.py

```diff
@@ -312,26 +397,34 @@ def on_key_request(self, messages):
 
     def check_key_response(self, messages):
         for message in messages:
-            if not message.source.startswith(u"circuit_"):
```

Comment:
That must have been caused by something else then, I would verify this using the request cache object to verify.